src/predicate/boxed.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2017, Nick Stevens <nick@bitcurry.com>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/license/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Predicate that can wrap other dynamically-called predicates in an
+//! easy-to-manage type.
+
+use std::fmt;
+
+use Predicate;
+
+/// `Predicate` that wraps another `Predicate` as a trait object, allowing
+/// sized storage of predicate types.
+pub struct BoxPredicate<T: ?Sized>(Box<Predicate<Item = T> + Send + Sync>);
+
+impl<T> fmt::Debug for BoxPredicate<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("BoxPredicate").finish()
+    }
+}
+
+impl<T> fmt::Display for BoxPredicate<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "BoxPredicate")
+    }
+}
```

Comment:
I've never implemented Display and Debug generically. Is it possible to display the contents of the box generically a well for the types that implement it?